src/structures/GuildMember.js

```diff
@@ -302,6 +302,17 @@ class GuildMember extends Base {
     return this.manageable && (this.guild.me?.permissions.has(Permissions.FLAGS.MODERATE_MEMBERS) ?? false);
   }
 
+  /**
+   * Whether this member is already under time out
+   * @returns {boolean}
+   */
+  isTimedOut() {
+    return (
+      typeof this.communicationDisabledUntilTimestamp === 'number' &&
+      this.communicationDisabledUntilTimestamp > Date.now()
+    );
+  }
```

Comment:
The `typeof` check is not needed here, simply `this.communicationDisabledUntilTimestamp > Date.now()` is enough. `null` will convert to `0`, so this is fine.

Such a simple check also makes me wonder if this should really be implemented, to be honest.

Reply:
> The `typeof` check is not needed here, simply `this.communicationDisabledUntilTimestamp > Date.now()` is enough. `null` will convert to `0`, so this is fine.

true, but it's always better to be on the safe side

> Such a simple check also makes me wonder if this should really be implemented, to be honest.

concise, succinct, self-descriptive, and of course narrows down `communicationDisabledUntilTimestamp` and `communicationDisabledUntil` as non-null properties like you pointed out below.
<sub>(I may or may not have used a dictionary while writing this down 🐱)</sub>

Reply:
> but it's always better to be on the safe side

It's literally ECMAScript's spec and it cannot change (without breaking the web, which they actively avoid).